src/httpfetch.h

```diff
@@ -48,12 +57,25 @@ struct HTTPFetchRequest
 
 	// Indicates if this is multipart/form-data or
 	// application/x-www-form-urlencoded.  POST-only.
+	// Data goes to post_fields!!!
 	bool multipart = false;
 
+	//  The Method to use default = GET
+	//  Avaible methods GET, POST, PUT, DELETE
+	HttpMethod method = HTTP_GET;
+
+	// Fields of the request
+	StringMap fields;
+
+	// Raw data of the request overrides ields
+	std::string raw_data;
+
+	// Deprecated for compatibility overrides method
 	// POST fields.  Fields are escaped properly.
 	// If this is empty a GET request is done instead.
 	StringMap post_fields;
 
+	// Deprecated for compatibility overrides method
```

Comment:
You can just remove this, and handle the deprecation it in the Lua API

Reply:
Yeah did that is now handled by the API